tiledb/sys/build/command.rs

```diff
@@ -0,0 +1,57 @@
+use std::thread;
+use std::time::Duration;
+
+use subprocess as sp;
+
+use crate::error::{Error, Result};
+
+pub fn run(cmd: &[&str], input: Option<&str>) -> Result<()> {
+    // Execute our Git command
+    let stdin = if input.is_some() {
+        sp::Redirection::Pipe
+    } else {
+        sp::Redirection::None
+    };
+
+    let mut git = sp::Popen::create(
+        cmd,
+        sp::PopenConfig {
+            stdin,
+            stdout: sp::Redirection::Pipe,
+            stderr: sp::Redirection::Pipe,
+            ..Default::default()
+        },
+    )
+    .map_err(|e| {
+        Error::Popen(format!("Spawning command: {}", cmd.join(" ")), e)
+    })?;
+
+    // Obtain the output from the standard streams.
+    let (out, err) = git.communicate(input).map_err(|e| {
+        Error::IO(format!("Executing command: {}", cmd.join(" ")), e)
+    })?;
+
+    // Wait for git to finish executing
+    loop {
+        let result = git.poll();
```

Comment:
Probably because I didn't take the time to read the docs very thoroughly. I happened to stumble over a good example of using `std::process::Command` this morning so I've since switched everything to that and its turned out cleaner.